skbio/io/format/stockholm.py

```diff
@@ -476,8 +476,9 @@ def add_gf_metadata(self, feature_name, feature_data):
             self._metadata[feature_name][tree_id] = (trees[tree_id] +
                                                      feature_data)
         elif feature_name in self._metadata:
+            padding = '' if self._metadata[feature_name].endswith(' ') else ' '
```

Comment:
Since we only want to add a space if the line doesn't end in whitespace, use `self._metadata[feature_name][-1].isspace()` which will handle space and other whitespace characters.